x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/LucenePushdownPredicates.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.optimizer.rules.physical.local;
+
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.expression.FieldAttribute;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.stats.SearchStats;
+
+/**
+ * When deciding if a filter or topN can be pushed down to Lucene, we need to check a few things on the field.
+ * Exactly what is checked depends on the type of field and the query. For example, we have the following possible combinations:
+ * <ol>
+ *     <li>A normal filter on a normal field will be pushed down using SingleValueQuery to remove multi-valued results,
+ *         and this requires knowing if the field is indexed and has doc-values.</li>
+ *     <li>A filter using a spatial function will allow multi-valued fields and we only need to know if the field is indexed,
+ *         and do not need doc values.</li>
+ *     <li>A TopN will be pushed down if the field is indexed and has doc values.</li>
+ *     <li>Filters with TEXT fields can only be pushed down if the TEXT field has a nested KEYWORD field,
+ *         referred to here as IdenticalDelegate. This that this is related to normal ES|QL predicates,
+ *         not the full-text search provided by the MATCH and QSTR functions, which are pushed down separately.</li>
+ * </ol>
+ */
+public interface LucenePushdownPredicates {
```

Comment:
Do we still need the default implementation ? I wonder how out of sync it will become with respect to SearchStats.

Separately there's a lot in this class - it defines both an interface, its implementations and acts as a factory so I suggest decomposing it:
1. have FieldAttribute in the name since all methods resolve around that - e.g. LucenePushDown, LuceneFieldInfo,  etc...
2. Move the two implementation into a Xs class such as LucenePushDowns, LuceneFieldInfos,  (like we do for DataType and DataTypes) :  LucenePushDowns.DEFAULT or LucenePushDowns.from(SearchStats)

Reply:
The recent work on MATCH has removed one of the three usages of the DEFAULT implementation. One other is test-only so could be refactored away, but the remaining use persists. It is in the `ComputeEngine.startComputeOnDataNodes` method, which creates a filter on `@timestamp` based on the current plan before we even know the separate data nodes, or individual SearchExecutionContexts to be used for creating the `SearchStats`. I'm maintaining the exact same behaviour here (ie. I did not want to change anything here, other than naming), although perhaps we would want to revisit this later. If so, we should do that in another PR.

(The above mentioned test-only usage is in `FilterTests`, but if we got rid of the ComputeEngine usage, we could mock SearchStats in FilterTests to get around that usage too, again a task for another PR if we desire it).

Regarding splitting the file into multiple files, I actually prefer the current approach because it is a simple interface with two simple implementations, and I think it is much easier to reason about with all three in the same file (which remains a very small file).